app-modules/service-management/src/Filament/Resources/ProductResource/Pages/ManageProductLicenses.php

```diff
@@ -0,0 +1,189 @@
+use Filament\Forms\Components\Group;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Checkbox;
+use Filament\Forms\Components\Textarea;
+use Filament\Tables\Actions\EditAction;
+use Filament\Tables\Actions\ViewAction;
+use Filament\Tables\Columns\TextColumn;
+use Filament\Forms\Components\TextInput;
+use Filament\Forms\Components\DatePicker;
+use Filament\Tables\Actions\CreateAction;
+use Filament\Tables\Actions\DeleteAction;
+use Filament\Infolists\Components\Section;
+use App\Filament\Tables\Columns\MaskColumn;
+use Filament\Infolists\Components\TextEntry;
+use Filament\Tables\Actions\BulkActionGroup;
+use Filament\Tables\Actions\DeleteBulkAction;
+use Filament\Resources\Pages\ManageRelatedRecords;
+use App\Filament\Infolists\Components\MaskTextEntry;
+use AidingApp\ServiceManagement\Filament\Resources\ProductResource;
+
+class ManageProductLicenses extends ManageRelatedRecords
+{
+    protected static string $resource = ProductResource::class;
+
+    protected static string $relationship = 'product_licenses';
+
+    protected static ?string $navigationLabel = 'Licenses';
+
+    protected static ?string $breadcrumb = 'License';
+
+    protected static ?string $navigationIcon = 'heroicon-o-key';
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                TextInput::make('license')
+                    ->label('License')
+                    ->string()
+                    ->required(),
+                Select::make('assigned_to')
+                    ->relationship(
+                        name: 'contact',
+                        titleAttribute: 'full_name',
+                    )
+                    ->label('Assigned To')
+                    ->native(false)
+                    ->preload()
+                    ->searchable(),
+                Textarea::make('description')
+                    ->label('Description')
+                    ->string()
+                    ->columnSpanFull(),
+                DatePicker::make('start_date')
+                    ->label('Start Date')
+                    ->displayFormat('d-m-Y')
+                    ->closeOnDateSelection()
+                    ->required()
+                    ->native(false)
+                    ->live()
+                    ->afterStateUpdated(
+                        fn (Get $get, Set $set) => $get('start_date') > $get('expiration_date') ? $set('expiration_date', '') : ''
+                    ),
+                Group::make([
+                    Checkbox::make('license_does_not_expire')
+                        ->label('License Does Not Expire')
+                        ->live()
+                        ->afterStateHydrated(function (Set $set, $state, $record) {
+                            if ($record && is_null($record->expiration_date)) {
+                                $set('license_does_not_expire', true);
+                            }
+                        }),
```

Comment:
@danharrin is this the best way to get this value initially? I feel like a `getStateUsing` or something like that may make more sense?

Reply:
All good